ci/check_and_fix_format.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -e
+
+function fix {
+  set +e
+  ci/do_ci.sh fix_format
+  ci/do_ci.sh fix_spelling
+  ci/do_ci.sh fix_spelling_pedantic
+  echo "Format check faild, try apply following patch to fix:"
+  git diff HEAD | tee /build/fix_format.diff
+
+  exit 1
+}
+
+trap fix ERR
```

Comment:
i'm unclear on what this is doing (and I thought i knew bash a little :) can you just add some comments?

Reply:
added.